app/connectors/ApplicationConnector.scala

```diff
@@ -62,4 +62,11 @@ trait ApplicationConnector {
   def fetchApplication(applicationId: String)(implicit hc: HeaderCarrier): Future[ApplicationWithHistory] = {
     http.GET[ApplicationWithHistory](s"$applicationBaseUrl/gatekeeper/application/$applicationId")
   }
+
+  def fetchAllApplicationsBySubscription(apiContext: Option[String])(implicit hc: HeaderCarrier): Future[Seq[ApplicationResponse]] = {
+    http.GET[Seq[ApplicationResponse]](s"$applicationBaseUrl/application?subscribesTo=${apiContext.getOrElse("")}")
+      .recover {
+        case e: Upstream5xxResponse => throw new FetchApplicationsFailed
+      }
+  }
```

Comment:
I would not assume that an application should treat: 

```
/path?query=
```

The same as 
```
/path
```
Which seems to be your intention here?  I can see the motivation for it below, because it allows convenient filtering. 

The Connectors should be fairly dumb/agnostic of the service use-cases below, though, so I'd suggest two different methods: 

```
def fetchAllApplicationsBySubscription(apiContext: String)(implicit hc: HeaderCarrier) = ???

def fetchAllApplications()(implicit hc: HeaderCarrier) = ??? 
```